reconcile/test/test_openshift_saas_deploy.py

```diff
@@ -86,3 +93,130 @@ def test_compose_console_url_with_long_saas_name(
         f"Pipeline name o-saas-deploy-{saas_name} is longer than 56 characters"
         == str(e.value)
     )
+
+
+def new_slack_api() -> slack_api.SlackApi:
+    api = slack_api.SlackApi(
+        workspace_name="test",
+        token="test-token",
+        init_usergroups=False,
+    )
+
+    def intercepted_slack_chat_post_message(message: str) -> str:
+        # Refer to the test_slack_notify_skipped_success method for a
+        # description of message_type
+        global message_type
```

Comment:
Don't use global var in test, a better way to write tests for `slack_notify` is like this:

first clarity what to test, example `test_slack_notify_skipped_success`

* Given `ri` has no errors
* When call `slack_notify` with `test_slack_notify_skipped_success: True`
* Then `slack.chat_post_message` should not be called

Translate that to code is this:

```python
def test_slack_notify_skipped_success():
    api = create_autospec(slack_api.SlackApi)

    slack_notify(
        saas_file_name="test-slack_notify--skipped-success.yaml",
        env_name="test",
        slack=api,
        ri=openshift_resource.ResourceInventory(),
        console_url="https://test.local/console",
        in_progress=False,
        skip_successful_notifications=True,
    )

    api.chat_post_message.assert_not_called()
```

Similar example for `test_slack_notify_unskipped_success`:

```python
def test_slack_notify_unskipped_success():
    api = create_autospec(slack_api.SlackApi)

    slack_notify(
        saas_file_name="test-slack_notify--unskipped-success.yaml",
        env_name="test",
        slack=api,
        ri=openshift_resource.ResourceInventory(),
        console_url="https://test.local/console",
        in_progress=False,
        skip_successful_notifications=False,
    )

    api.chat_post_message.assert_called_once_with(
        ":green_jenkins_circle: SaaS file *test-slack_notify--unskipped-success.yaml* "
        "deployment to environment *test*: Success "
        "(<https://test.local/console|Open>)"
    )
```

Tests should verify the full expected behaviour, use full raw input and output(side effect) to cover all cases.

Reply:
Thanks for that. It felt really gross to use the `global` but Python is not my first (second, third, etc) language. This really gives me something to work with.

Reply:
@hemslo How does it look now?

Reply:
Nice! just need to cleanup commented code https://github.com/app-sre/qontract-reconcile/pull/3913/files/cf13bacf93706ba9107fc19dee89e874a83837c9..644a2d89f2da4a77d22d8d548f801bb78ccdd17f#diff-0f6dbc40d4600c4527755db5370c8c15906daa9c1269fe71e0de1add0243849eR174

Reply:
Oh! Well spotted.